federation/pkg/federation-controller/service/dns/dns.go

```diff
@@ -111,7 +116,14 @@ func NewServiceDNSController(client fedclientset.Interface, dnsProvider, dnsProv
 		},
 		&v1.Service{},
 		serviceSyncPeriod,
-		util.NewTriggerOnAllChanges(func(obj pkgruntime.Object) { d.workQueue.Add(obj) }),
+		util.NewTriggerOnAllChanges(func(obj pkgruntime.Object) {
+			switch value := obj.(type) {
+			case *v1.Service:
+				d.deliverService(value, 0, false)
```

Comment:
I realized that there was no need to change this. We can delay delivery only if there exist some error, otherwise the new events can directly be enqueued in work queue. So moved removed this change now.